scanner.py

```diff
@@ -427,117 +510,591 @@ def start_touch(self, gcmd, touch_settings, verbose):
+                    ):
+                        # Increase threshold_max by 3 steps above the current threshold, only if it hasn't been increased before
+                        if not has_increased_threshold_max:
+                            threshold_max = current_threshold + 5 * step
+                            has_increased_threshold_max = (
+                                True  # Set the flag to True after increasing
+                            )
+                            gcmd.respond_info(
+                                f"Good Threshold Found: {current_threshold}. Adjusting threshold_max to {threshold_max}"
+                            )
+
+                        # Run repeatability check with this threshold
+                        consistent_results = True
+                        for attempt in range(repeat_attempts):
+                            repeat_result = self.start_threshold_scan(
+                                gcmd, touch_settings, verbose
+                            )
+                            if not repeat_result["success"] or (
+                                repeat_result["standard_deviation"]
+                                > max_acceptable_std_dev
+                            ):
+                                gcmd.respond_info(
+                                    f"Repeat attempt {attempt + 1} failed for threshold {current_threshold}"
+                                )
+                                consistent_results = False
+                                break
+                            gcmd.respond_info(
+                                f"Repeat attempt {attempt + 1} successful with std dev: {repeat_result['standard_deviation']:.5f}"
+                            )
+
+                        # Save only successful repeat attempts in results
+                        result["consistent_results"] = (
+                            consistent_results  # Mark if it passed repeatability checks
+                        )
+                        result["standard_deviation"] = (
+                            repeat_result["standard_deviation"]
+                            if consistent_results
+                            else None
+                        )
+                        result["success"] = (
+                            repeat_result["success"] if consistent_results else False
+                        )
+                        results.append(result)  # Append the result for later comparison
+                        # If all repeat attempts succeeded, save the threshold
+                        if consistent_results and override == 0:
+                            best_threshold = current_threshold
+                            self._save_threshold(best_threshold)
+                            gcmd.respond_info(
+                                f"Confirmed Optimal Threshold Found: {best_threshold}"
+                            )
+                            break
```

Comment:
Obsolete message. We also mention it on line 877.

Reply:
done :D